physics/MP/NSSL/module_mp_nssl_2mom.F90

```diff
@@ -5929,6 +6078,104 @@ SUBROUTINE calc_eff_radius    &
          ENDIF
          ENDIF
 
+         IF ( present(t5) .and. flag_t5 ) THEN
+         
+         ! first: case when hail is off
+          
+         IF (  lhl < 1 .or. flag_t6 ) THEN
+         ! graupel only 
+           IF ( qx(mgs,lh) > Max(1.e-8,qxmin(lh)) ) THEN
+            ! Lambda for graupel
+            hwdn = xdn0(lh)
+            IF ( lvh > 1 ) THEN ! variable density
+              IF ( an(ix,jy,kz,lvh) > 1.e-30 ) THEN
+                hwdn = rho0(mgs)*qx(mgs,lh)/an(ix,jy,kz,lvh)
+              ENDIF
+            ENDIF
+            
+              lam_h = factor_h *((hwdn*cx(mgs,lh))/(qx(mgs,lh)*rho0(mgs)))**(1./3.)
```

Comment:
`1./3.` is used 125 times in the code. Many schemes define it as a constant at the top of the scheme and use it throughout to achieve a minor speed up. Something to consider.

Reply:
Yes, I have seen that done. I think the speedup would only be at -O0, since most compilers will precompute the 1./3. with any optimization (maybe even at 0, but I don't know). It's hard to outsmart modern compilers